src/http/client.cc

```diff
@@ -141,8 +141,17 @@ future<connection::reply_ptr> connection::recv_reply() {
                 throw std::system_error(ECONNABORTED, std::system_category());
             }
             if (parser.failed()) {
-                http_log.trace("Parsing response failed");
-                throw std::runtime_error("Invalid http server response");
+                http_log.warn("Parsing response failed.{}", http_log.level() == log_level::debug ? format(" Reason: {}", parser.error_message()) : "");
+                // Parsing the response failed, potentially due to a server-side bug or
+                // a transient error. Retrying might resolve the issue. To trigger a retry,
+                // return EPIPE to the caller. The original exception is retained as a
+                // nested exception so that, in case retries keep failing, the caller can
+                // identify the actual cause unrelated to the EPIPE error.
+                try {
+                    throw std::system_error(EPIPE, std::system_category());
```

Comment:
Let's better throw a dedicated error-parsing-http-server-response-line-exception thing. It will carry the error message on board, and it will be explicitly caught by the restart checking code.

Reply:
and then we just want to add another condition to the `if` [here](https://github.com/scylladb/seastar/blob/master/src/http/client.cc#L339) instead of reusing existing nested exception thingy?

Reply:
Yes. It's not great to masquerade parsing error with EPIPE exception, because it's not such